### ISSUE TITLE
PyInstaller --uac-admin not working without Specified Icon

### ISSUE THREAD

**Issue by StephenNorCalControls** [pyinstaller/pyinstaller#4752]
+++ ONLY TEXT +++ DO NOT POST IMAGES +++

## Description of the issue
Using the --uac-admin option or specifying uac_admin=True in the spec file will not prompt windows to open the exe as an admin unless an icon is also specified (I suspect that it just has to be some manifest-based resource)

My spec file: (names changed for anonymity)
```
# -*- mode: python ; coding: utf-8 -*-

block_cipher = None

a = Analysis(['AppName.py'],
             pathex=['C:\\Users\\MyUser\\PycharmProjects\\AppName'],
             binaries=[],
             datas=[("./dist/ExtraExecutable.exe", "/dist"), ("./favicon.ico", ".")],
             hiddenimports=[],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)

pyz = PYZ(a.pure, a.zipped_data,
          cipher=block_cipher)

exe = EXE(pyz,
          a.scripts,
          a.binaries,
          a.zipfiles,
          a.datas,
          [],
          name='AppName',
          debug=False,
          bootloader_ignore_signals=False,
          strip=False,
          upx=True,
          upx_exclude=[],
          runtime_tmpdir=None,
          console=True,
          uac_admin=True,
          icon='favicon.ico')
```

* Output of `pyinstaller --version`: 3.6
* Version of Python: 3.7.6
* Platform: Windows 10 Pro 1909 (Build 18363.657)
* Did you also try this on another platform? Does it work there? Untested

* try the latest development version, using the following command:

```shell
pip install https://github.com/pyinstaller/pyinstaller/archive/develop.zip
```

* follow *all* the instructions in our "If Things Go Wrong" Guide
  (https://github.com/pyinstaller/pyinstaller/wiki/If-Things-Go-Wrong) and

### Make sure [everything is packaged correctly](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly)

  * [x] start with celan installation
  * [x] use the latest development version
  * [x] Run your frozen program **from a command window (shell)** — instead of double-clicking on it
  * [x] Package your program in **--onedir mode**
  * [x] Package **without UPX**, say: use the option `--noupx` or set `upx=False` in your .spec-file
  * [x] Repackage you application in **verbose/debug mode**. For this, pass the option `--debug` to `pyi-makespec` or `pyinstaller` or use `EXE(..., debug=1, ...)` in your .spec file.

### A minimal example program which shows the error

```
1. Create a file and only add: `print("Hello World!")`
2. Bundle it with PyInstaller, which should create a spec file
3. Try again using either the spec file or command-line options adding --uac-admin or uac_admin=True respectively.
4. Notice how the generated exe will not be required to open as admin and no UAC prompt will be generated.
5. Add an icon to the project, using either icon=./somefile.ico or icon='./somefile.ico' to the command line or spec file respectively.
6. Notice how the output of the PyInstaller command will mention that a manifest needed to be appended because a resource changed and the generated exe will be required to open as admin (and have an icon now)

What happens: the generated exe is not required to open as an admin unless an icon file is specified

What should happen: the generated exe should be run as admin regardless of icon inclusion.

Output exe is the same in both the latest release as well as the development build given above.

“Minimal“ means: remove everything from your code which is not relevant for this bug,
esp. don't use external programs, remote requests, etc.
A very good example is https://gist.github.com/ronen/024cdae9ff2d50488438. This one helped
us reproducing and fixing a quite complex problem within approx 1 hour.
```

### Stacktrace / full error message

```
N/A
```

**Comment by Legorooj** [comment]
Can you try on another machine? I can't reproduce this.

**Comment by StephenNorCalControls** [comment]
Hi, I tested on a completely different system with the same result. Are you using anything like a virtual environment? I tried both with a venv and without, and got the same result. Attaching an icon file correctly builds the exe with a manifest, and consequently appends the resource that is responsible for signalling to the windows OS that the executable needs to run as admin. If uac-admin is specified without an icon, then no manifest is ever attached and thus the exe simply runs without admin privileges.

**Comment by Legorooj** [comment]
@StephenNorCalControls I'll have another go.

**Comment by Legorooj** [comment]
@StephenNorCalControls I can't replicate this.

Could you try cloning the git repo, building the bootloader, and then installing before trying again?

**Comment by StephenNorCalControls** [comment]
After cloning the repo and building it, I still see the same issue. Attached are the output logs from two builds with slightly different parameters.

> (venv) C:\Users\Stephen\Desktop\pyinstaller-develop>PyInstaller Test.py --onefile --uac-admin --icon=icon.ico
> 93 INFO: PyInstaller: 4.0.dev0+80d5078081
> 93 INFO: Python: 3.7.6
> 94 INFO: Platform: Windows-10-10.0.18362-SP0
> 94 INFO: wrote C:\Users\Stephen\Desktop\pyinstaller-develop\Test.spec
> 97 INFO: UPX is not available.
> 99 INFO: Extending PYTHONPATH with paths
> ['C:\\Users\\Stephen\\Desktop\\pyinstaller-develop',
>  'C:\\Users\\Stephen\\Desktop\\pyinstaller-develop']
> 100 INFO: checking Analysis
> 111 INFO: checking PYZ
> 121 INFO: checking PKG
> 123 INFO: Building because C:\Users\Stephen\Desktop\pyinstaller-develop\build\Test\Test.exe.manifest changed
> 123 INFO: Building PKG (CArchive) PKG-00.pkg
> 1172 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
> 1174 INFO: Bootloader C:\Users\Stephen\Desktop\pyinstaller-develop\venv\lib\site-packages\pyinstaller-4.0.dev0+80d5078081-py3.7.egg\PyInstaller\bootloader\Windows-64bit\run.exe
> 1174 INFO: checking EXE
> 1174 INFO: Building because icon changed
> 1174 INFO: Building EXE from EXE-00.toc
> 1176 INFO: Copying icons from ['asian_food_bowl_vegetables_icon_142426.ico']
> 1177 INFO: Writing RT_GROUP_ICON 0 resource with 20 bytes
> 1177 INFO: Writing RT_ICON 1 resource with 47032 bytes
> 1180 INFO: Updating manifest in C:\Users\Stephen\Desktop\pyinstaller-develop\build\Test\run.exe.c2nsv1g_
> 1181 INFO: Updating resource type 24 name 1 language 0
> 1183 INFO: Appending archive to EXE C:\Users\Stephen\Desktop\pyinstaller-develop\dist\Test.exe
> 1188 INFO: Building EXE from EXE-00.toc completed successfully.

> (venv) C:\Users\Stephen\Desktop\pyinstaller-develop>PyInstaller Test.py --onefile --uac-admin
> 108 INFO: PyInstaller: 4.0.dev0+80d5078081
> 108 INFO: Python: 3.7.6
> 109 INFO: Platform: Windows-10-10.0.18362-SP0
> 109 INFO: wrote C:\Users\Stephen\Desktop\pyinstaller-develop\Test.spec
> 112 INFO: UPX is not available.
> 113 INFO: Extending PYTHONPATH with paths
> ['C:\\Users\\Stephen\\Desktop\\pyinstaller-develop',
>  'C:\\Users\\Stephen\\Desktop\\pyinstaller-develop']
> 114 INFO: checking Analysis
> 126 INFO: checking PYZ
> 135 INFO: checking PKG
> 136 INFO: Building because C:\Users\Stephen\Desktop\pyinstaller-develop\build\Test\Test.exe.manifest changed
> 136 INFO: Building PKG (CArchive) PKG-00.pkg
> 1216 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
> 1218 INFO: Bootloader C:\Users\Stephen\Desktop\pyinstaller-develop\venv\lib\site-packages\pyinstaller-4.0.dev0+80d5078081-py3.7.egg\PyInstaller\bootloader\Windows-64bit\run.exe
> 1218 INFO: checking EXE
> 1218 INFO: Building because icon changed
> 1218 INFO: Building EXE from EXE-00.toc
> 1219 INFO: Appending archive to EXE C:\Users\Stephen\Desktop\pyinstaller-develop\dist\Test.exe
> 1224 INFO: Building EXE from EXE-00.toc completed successfully.

In the first  instance, I specify the icon file, and in the output logs it reports that the manifest was changed and had to be  added to the executable. In the second instance, even with `--uac-admin` specified, it correctly detects that the icon changes, but it did not update the manifest, and the resultant exe has no prompt to run through uac. 

I tried building them in both orders (with icon then without icon and vice versa) to the same effect.

**Comment by Legorooj** [comment]
I didn't use `--onefile`. I'll try that. Try building with `--clean`?

**Comment by StephenNorCalControls** [comment]
I added `--clean` to both build strings and got the same result. It doesn't seem to change the outcome.

**Comment by Legorooj** [comment]
@StephenNorCalControls verified.

> (I suspect that it just has to be some manifest-based resource)

You're probably correct.

This also only occurs with `--onefile`. `-D/--onedir` works fine when running 80d5078081.

**Comment by Legorooj** [comment]
Tracked this failiure quite easily once I got round to it. Fixed in f11fb8b by adding `uac_admin` and `uac_uiaccess` to the manifest embedding trigger.
